esrally/track/params.py

```diff
@@ -197,25 +198,39 @@ def __init__(self, indices, params):
                                            (id_conflicts, action_metadata))
 
         self.pipeline = params.get("pipeline", None)
-        try:
-            self.bulk_size = int(params["bulk-size"])
-            if self.bulk_size <= 0:
-                raise exceptions.InvalidSyntax("'bulk-size' must be positive but was %d" % self.bulk_size)
-        except KeyError:
-            raise exceptions.InvalidSyntax("Mandatory parameter 'bulk-size' is missing")
-        except ValueError:
-            raise exceptions.InvalidSyntax("'bulk-size' must be numeric")
-
-        try:
-            self.batch_size = int(params.get("batch-size", self.bulk_size))
-            if self.batch_size <= 0:
-                raise exceptions.InvalidSyntax("'batch-size' must be positive but was %d" % self.batch_size)
-            if self.batch_size < self.bulk_size:
-                raise exceptions.InvalidSyntax("'batch-size' must be greater than or equal to 'bulk-size'")
-            if self.batch_size % self.bulk_size != 0:
-                raise exceptions.InvalidSyntax("'batch-size' must be a multiple of 'bulk-size'")
-        except ValueError:
-            raise exceptions.InvalidSyntax("'batch-size' must be numeric")
+        def handler_size_number(params_size, params_key):
+            #regexp the size with unit
+            if_with_unit = re.findall(r'(\d+)([kKmMgGbB]+)', params_size)
+            # not with unit
+            if len(if_with_unit) == 0:
+                try:
+                    temp_size = int(params_size)
+                    if temp_size <=0:
+                        raise exceptions.InvalidSyntax("'%s' must be positive but was %d" % (params_key, params_size))
+                except ValueError:
+                    raise exceptions.InvalidSyntax("'%s' must be numeric"% (params_key) )
+            else:
+                temp_size = if_with_unit[0][0]
+                temp_unit = if_with_unit[0][1].upper()
+                if temp_unit == "GB":
+                    temp_size = int(float(temp_size) * 1024 * 1024 * 1024)
+                elif temp_unit == "MB":
+                    temp_size = int(float(temp_size) * 1024 * 1024)
+                elif temp_unit == "KB":
+                    temp_size = int(float(temp_size) * 1024)
+                else:
+                    raise AttributeError("'%s' unit '%s' must be KB/MB/GB" % (params_size, temp_size))
+            return temp_size
+
+        for key in ["bulk-size", "batch_size"]:
+            if key not in params:
+                raise KeyError("Mandatory parameter '%s' is missing" % (key))
+        self.bulk_size = handler_size_number(params["bulk-size"], "bulk-size")
```

Comment:
Hmm, suppose the user has specified a bulk size of "500 KB". `handler_size_number()` will return 500 * 1.024 = 512.000 (bytes). But the code down the line (specifically `params.IndexDataReader` and `runner.BulkIndex` has no clue that it should handle bytes now. So what will happen instead is that `params.IndexDataReader` will read 512.000 lines from the source file and `params.IndexDataReader` will report a bulk size of 512.000 docs. So we need to represent the unit too + enable `params.IndexDataReader` to read lines until the limit is exhausted. 

Additionally, you need to adapt `params.Slice` which implements the skipping logic (based on lines currently). A little bit of background on this skipping logic: If a user specifies that N clients should be used for bulk-indexing (where N > 1), Rally will create N sub-processes and each sub-process will read one-Nth of the source file. The skipping logic with the file offset table is necessary because all clients should start roughly at the same time to maximize load on the cluster and my experiments have shown that without this skipping logic it takes clients up to a minute to skip to the relevant slice of a large file (even on a very fast SSD!). I hope it makes sense now with this description.

Finally, you should never reach this point in the code due to a JSON schema validation error because `bulk-size` is declared as `integer` in `resources/track-schema.json`. I guess we need to remove the specification of `bulk-size` entirely there and reimplement the necessary constraint checks (only positive integer values or number + "B/KB/MB/GB" values are allowed). Btw, I think we should not support GB bulk sizes as it makes no practical sense.